File: leetcode094.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if root == None:
            return []
        res = []
        self.recursiveTraversal(root, res)
        return res

    def recursiveTraversal(self, root, res):

        if root.left != None:
            self.recursiveTraversal(root.left, res)

        res.append(root.val)
        if root.right != None:
            self.recursiveTraversal(root.right, res)

        return 

File: leetcodec++/leetcode094.cpp
//
// Created by 黄敏辉 on 2019-07-30.
//

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        inorder(root, res);
        return res;
    }

    void inorder(TreeNode* root, vector<int>& res){
        if (!root)
            return;
        inorder(root->left, res);
        res.push_back(root->val);
        inorder(root->right, res);
    }
};